testing/unit/rpc/test_block_info.py

```diff
@@ -172,6 +173,544 @@
+                            'payload': 'CtICCAESgAFhZGM0ZGY1NzdlNDM0ZjUzNjBkYTk2YTkzYjU0MDRmNjFiMzkyZDc2MjdhNjlhYzMy'
+                                       'Y2U0YTYxODg5ZWE4NTEzMTk4Yzk2YTU4ODdlYTViYjM2NWU0ZjE1MzM3YjIzMWZhOWY2ZTRhMWNj'
+                                       'ZTcwZDU3NTk4Yzg3YzRhYjVjNTQ0ZRpCMDJjMTcyZjlhMjc1MTJjMTFlMmQ0OWZkNDFhZGJjYjIx'
+                                       'NTE0MDNiZDE1ODJlOGNkOTRhNTE1Mzc3OWMyMTA3MDkyIoABMzM3NDkyZDE5MDY3OGZiZTAxMWEy'
+                                       'ZDA5MzcwNzdhYTk0OWQ1ODJlZWQ5Mzc4MWYzMmI0OTMyMDU1Zjc5MDEwNzNmODk3YTViNzA3ZWI4'
+                                       'M2NkMTU1N2VmNGM3MGUzYzA5ZTEzNjhlZDlmODk3MTIwMWVlOGE0MDljYTQwN2UxNDMoos/o5AU='
+                        }
+                    ]
+                },
+            ],
+            'header': {
+                'batch_ids': [
+                    '50898544a24cc09fbefb24402bcea4ff15801d6de0bb7d1b0af1402038f24e8e'
+                    '1f1bf25c782b1711c58547ee43573deb0f94656ade83ed8b70df41b7fcb4f2fe',
+                    'e7ce94ffd2ece31d69c123a9666158c4cdaca7522a41c87d6088364d309b8e2a'
+                    '50f070e4b91b4f17796dc5248b344d69ed99bd4ee6aaa384a7d76a43a7256726'
+                ],
+                'block_num': '2',
+                'consensus': 'RGV2bW9kZbBTi7/XqZI1BSUPdd+zjPl2xPTcd8WjLDB12DVIRWNV',
+                'previous_block_id': '337492d190678fbe011a2d0937077aa949d582eed93781f32b4932055f790107'
+                                     '3f897a5b707eb83cd1557ef4c70e3c09e1368ed9f8971201ee8a409ca407e143',
+                'signer_public_key': '02c172f9a27512c11e2d49fd41adbcb2151403bd1582e8cd94a5153779c2107092',
+                'state_root_hash': 'b865cf3ea5e5dc0698c37e464e65a5ac417fe5c643109c8c49d465d152dc027f'
+            },
+            'header_signature': '92fa987b4f163b43d9f85641bb8ccf018022d3a9e2445e3c721e0b3cdfa83c42'
+                                '65a21c74be9f6a732f75fcab1832637fa98816cd7fb43e42b9f609ffae1136fd'
+        }
+    }
+
+    mock_fetch_block = mocker.patch('remme.shared.router.Router.list_blocks')
+    mock_fetch_block.return_value = return_async_value(expected_result)
+
+    result = await list_blocks(request_)
+
+    assert expected_result == result
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('invalid_ids', (0, 1, ['12345'], [0, ], ['12345', 12345], [True, False]))
+async def test_list_blocks_with_invalid_ids(request_, invalid_ids):
+    """
+    Case: list blocks with invalid parameter ids.
+    Expect: header signature is not of a blockchain token type error message.
+    """
+    request_.params = {
+        'ids': invalid_ids,
+    }
+
+    with pytest.raises(RpcInvalidParamsError) as error:
+        await list_blocks(request_)
+
+    assert 'Header signature is not of a blockchain token type.' == error.value.message
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize('invalid_blocks_ids', (['12345', 0, 12345, True]))
```

Comment:
@Alladin9393, move these parameters to the above test, because tests are identical.

Reply:
@anastasiia-bilova, done.